### ISSUE TITLE
Question on ./model/dgaf.py

### ISSUE THREAD

**Issue by women1995** [zhwzhong/DAGF#1]
Good job! Thank you for your excellent work! 
I have a question about the Kernel Generation, i.e., what the meaning of the following code in the FuseBlock module.

-         fuse_kernel = torch.tanh(fuse_kernel) / (self.aff_scale_const + 1e-8)
- 
-        abs_kernel = torch.abs(fuse_kernel)
-        abs_kernel_sum = torch.sum(abs_kernel, dim=1, keepdim=True) + 1e-4
- 
-         abs_kernel_sum[abs_kernel_sum < 1.0] = 1.0
- 
-         fuse_kernel = fuse_kernel / abs_kernel_sum

 Could you please provide more information about this? Thank you for your kind consideration of this request.

**Comment by zhwzhong** [comment]
Hello, thank you for attention, 
    Its function is to let the sum of the learned kernel weights to one, i.e.,  \sum_{i=1}^{k^2} w_1 = 1

**Comment by women1995** [comment]
Thanks for your answer.  
I have one more question, i.e.,  why use this form for kernel normalization rather than softmax functions? What is the physical meaning of this form?

**Comment by zhwzhong** [comment]
Hello, 
   The kernel normalization is a common way for the guided filter, and you can use any kind of function (e.g., softmax) you like to achieve this goal.  I suggest that you read some papers about guided filters to find more reasonable explanations.

Best regards,
Zhiwei Zhong

**Comment by zhwzhong** [comment]
If you have any questions you can send me an email  (zhwzhong@hit.edu.cn) or add my WeChat for more convenient communication.

Thanks.
